virtualizarr/parsers/zarr.py

```diff
@@ -84,9 +84,10 @@ def get_metadata(zarr_array: zarr.AsyncArray[Any]) -> ArrayV3Metadata:
     zarr_format = zarr_array.metadata.zarr_format
 
     if zarr_format == 2:
-        # TODO: Once we want to support V2, we will have to deconstruct the
-        # zarr_array codecs etc. and reconstruct them with create_v3_array_metadata
-        raise NotImplementedError("Reading Zarr V2 currently not supported.")
+        # only import this if needed because it's private zarr internals which are likely unstable
+        from zarr.metadata.migrate_v3 import _convert_array_metadata
+
+        return _convert_array_metadata(zarr_array.metadata)
```

Comment:
fyi @d-v-b I think this is the piece of API from https://github.com/zarr-developers/zarr-python/pull/3257 that we would ideally like to be exposed as public and stable in zarr-python.